lib/stripe/list_object.rb

```diff
@@ -12,10 +22,35 @@ def [](k)
       end
     end
 
+    # Iterates through each resource in the page represented by the current
+    # `ListObject`.
+    #
+    # Note that this method makes no effort to fetch a new page when it gets to
+    # the end of the current page's resources. See also +each_with_all_pages+.
     def each(&blk)
       self.data.each(&blk)
     end
 
+    # Iterates through each resource in all pages, making additional fetches to
+    # the API as necessary.
+    #
+    # Note that this method will make as many API calls as necessary to fetch
+    # all resources. For more granular control, please see +each+ and
+    # +next_page+.
+    def each_with_all_pages(&blk)
```

Comment:
As currently written, if you pass a block to this, it won't actually call it. I think you want something like this:

```
def each_with_all_pages
  return enum_for(:each_with_all_pages) unless block_given?
  page = self
  loop do
    page.each do |resource|
      yield resource
    end

    page = page.next_page
    break if page.empty?
  end
end
```

(And should also add some tests that call this with a block.)